hydromt_sfincs/utils.py

```diff
@@ -623,6 +625,97 @@ def read_geoms(fn: Union[str, Path]) -> List[Dict]:
     return feats
 
 
+def write_drn(fn: Union[str, Path], gdf_drainage: gpd.GeoDataFrame, fmt="%.4f") -> None:
+    """Write structure files from list of dictionaries.
+
+    Parameters
+    ----------
+    fn : str, Path
+        Path to structure file.
+    drainage : gpd.GeoDataFrame
+        Dataframe with drainage structure parameters and geometry.
+    fmt : str
+        Format for float values.
+    """
+
+    # expected columns for drainage structures
+    col_names = [
+        "xsnk",
+        "ysnk",
+        "xsrc",
+        "ysrc",
+        "type",
+        "par1",
+        "par2",
+        "par3",
+        "par4",
+        "par5",
+    ]
+
+    gdf = copy.deepcopy(gdf_drainage)
+    # get geometry linestring and convert to xsnk, ysnk, xsrc, ysrc
+    gdf["xsnk"] = [ls.coords[0][0] for ls in gdf.geometry]
+    gdf["ysnk"] = [ls.coords[0][1] for ls in gdf.geometry]
+    gdf["xsrc"] = [ls.coords[1][0] for ls in gdf.geometry]
+    gdf["ysrc"] = [ls.coords[1][1] for ls in gdf.geometry]
```

Comment:
Should the source be the start point of each line and the sink the endpoint? If so this should still be adjusted in my suggestion below. 

Here you can also avoid the loops using
```suggestion
    endpoints = gdf.boundary.explode().unstack()
    gdf["xsnk"] = endpoints[0].x
    gdf["ysnk"] = endpoints[0].y
    gdf["xsrc"] = endpoints[1].x
    gdf["ysrc"] = endpoints[1].y
```